src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -48,27 +48,23 @@ public class UserManager<TUser> : IDisposable where TUser : class
         /// <param name="contextAccessor">The accessor used to access the <see cref="HttpContext"/>.</param>
         public UserManager(IUserStore<TUser> store,
             IOptions<IdentityOptions> optionsAccessor,
-            IPasswordHasher<TUser> passwordHasher,
-            IEnumerable<IUserValidator<TUser>> userValidators,
-            IEnumerable<IPasswordValidator<TUser>> passwordValidators,
-            ILookupNormalizer keyNormalizer,
-            IdentityErrorDescriber errors,
             IServiceProvider services,
-            ILogger<UserManager<TUser>> logger,
-            IHttpContextAccessor contextAccessor)
+            ILogger<UserManager<TUser>> logger)
         {
             if (store == null)
             {
                 throw new ArgumentNullException(nameof(store));
             }
             Store = store;
-            Options = optionsAccessor?.Value ?? new IdentityOptions();
-            _context = contextAccessor?.HttpContext;
-            PasswordHasher = passwordHasher;
-            KeyNormalizer = keyNormalizer;
-            ErrorDescriber = errors;
             Logger = logger;
+            Options = optionsAccessor?.Value ?? new IdentityOptions();
+
+            _context = services?.GetService<IHttpContextAccessor>()?.HttpContext;
+            PasswordHasher = services?.GetService<IPasswordHasher<TUser>>() ?? new PasswordHasher<TUser>();
+            KeyNormalizer = services?.GetService<ILookupNormalizer>() ?? new UpperInvariantLookupNormalizer(); ;
```

Comment:
I prefer to be explicit when it comes to dependencies. Now most the param doc-commments don't make sense. I wouldn't remove the doc-comments, because I think it's useful to explain which services can be replaced to change the behavior of the user manager. However, if we move to the service locator pattern, we don't have a standard way (i.e. `<param...`) to describe the dependencies and how they are used.

I also don't like hiding the default implementation of services. I guess it's not *really* hidden since the default implementation here matches those added by `AddIdentity`, but that just makes this less DRY. 

Newing up the default service implementation also isn't a good solution in the generaly case, even though it doesn't seem like it's an issue here, because things quickly get out of hand if there are transitive dependencies.

For services that really are optional, as opposed to having a known default implementation, I am more sympathetic to using the service locator pattern. Even then, there is an alternative, and that is to use `IEnumerable<TService>`. The only real downside I see to this is that the type implies that 0-many services can be provided when we really expect one service or none at all. UserManager doesn't seem to have any of the truly optional services though.